File: mobile/src/main/java/net/mm2d/dmsexplorer/log/Sender.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.log;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.Size;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public interface Sender {
    void logEvent(
            @NonNull @Size(min = 1L, max = 40L) String name,
            @Nullable Bundle params);
}

File: mobile/src/main/kotlin/net/mm2d/dmsexplorer/log/Sender.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.log

import android.os.Bundle
import androidx.annotation.Size

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
internal interface Sender {
    fun logEvent(
        @Size(min = 1L, max = 40L) name: String,
        params: Bundle?,
    )
}
